diff --git a/python-rhsm.spec b/python-rhsm.spec
index 872823d..e5fdf89 100644
--- a/python-rhsm.spec
+++ b/python-rhsm.spec
@@ -11,7 +11,7 @@
 
 Name: python-rhsm
 Version: 1.12.5
-Release: 1%{?dist}
+Release: 2%{?dist}
 
 Summary: A Python library to communicate with a Red Hat Unified Entitlement Platform
 Group: Development/Libraries
@@ -71,6 +71,9 @@ rm -rf %{buildroot}
 %attr(640,root,root) %{_sysconfdir}/rhsm/ca/*.pem
 
 %changelog
+* Thu Aug 28 2014 jesus m. rodriguez <jesusr@redhat.com> 1.12.5-2
+- Use distribution builder, setup 6.6 builds (jesusr@redhat.com)
+
 * Fri Jul 25 2014 jesus m. rodriguez <jesusr@redhat.com> 1.12.5-1
 - 1120431: Support for complex path matching. (bkearney@redhat.com)
 
diff --git a/rel-eng/packages/python-rhsm b/rel-eng/packages/python-rhsm
index 2deb3bb..6dc8b36 100644
--- a/rel-eng/packages/python-rhsm
+++ b/rel-eng/packages/python-rhsm
@@ -1 +1 @@
-1.12.5-1 ./
+1.12.5-2 ./
diff --git a/rel-eng/releasers.conf b/rel-eng/releasers.conf
index 8594eaf..9462ffe 100644
--- a/rel-eng/releasers.conf
+++ b/rel-eng/releasers.conf
@@ -1,34 +1,3 @@
-[yum-el6-x86_64]
-releaser = tito.release.YumRepoReleaser
-builder = tito.builder.MockBuilder
-builder.mock = epel-6-x86_64
-rsync = fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-6Server/x86_64/ fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-6/x86_64/
-
-[yum-el6-i386]
-releaser = tito.release.YumRepoReleaser
-builder = tito.builder.MockBuilder
-builder.mock = epel-6-i386
-rsync = fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-6Server/i386/ fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-6/i386/
-
-[yum-el5-x86_64]
-releaser = tito.release.YumRepoReleaser
-builder = tito.builder.MockBuilder
-builder.mock = epel-5-x86_64
-rsync = fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-5Server/x86_64/ fedorapeople.org:/srv/repos/candlepin/subscription-manager/epel-5/x86_64/
-
-[fedora]
-releaser = tito.release.FedoraGitReleaser
-branches = master f20 f19
-
 [rhel-6.6]
 releaser = tito.release.DistGitReleaser
 branches = rhel-6.6
-required_bz_flags = rhel-6.6.0+
-placeholder_bz = 1109802
-
-[rhel-7.1]
-releaser = tito.release.DistGitReleaser
-branches = rhel-7.1
-required_bz_flags = rhel-7.1.0+
-placeholder_bz = 1109810
-
diff --git a/rel-eng/tito.props b/rel-eng/tito.props
index b1125f2..6c86a16 100644
--- a/rel-eng/tito.props
+++ b/rel-eng/tito.props
@@ -1,4 +1,4 @@
 [globalconfig]
-default_builder = tito.builder.Builder
-default_tagger = tito.tagger.VersionTagger
+default_builder = tito.distributionbuilder.DistributionBuilder
+default_tagger = tito.tagger.ReleaseTagger
 
